### PR TITLE
Initial hunter packages for OpenCV, dlib, glog, csv-parser-cplusplus

### PR DESCRIPTION
**Pull request by headupinclouds**

### CODE REVIEW

File: cmake/projects/Eigen/hunter.cmake [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+# Load used modules
+include(hunter_add_version)
+include(hunter_add_version_start)
+include(hunter_download)
+include(hunter_pick_scheme)
+include(hunter_add_package)
+
+# Print version that we are trying to download (i.e. from 'config.cmake')
+hunter_add_version_start(Eigen)
```
Comment (ruslo) [inline cmake/projects/Eigen/hunter.cmake]: There is no such module anymore. This command not needed for now.

File: cmake/projects/Eigen/hunter.cmake [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# This is a header-like file, so include guards needed
```
Comment (ruslo) [inline cmake/projects/Eigen/hunter.cmake]: @headupinclouds Please leave your copyrights here

File: cmake/projects/csv-parser-cplusplus/hunter.cmake [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    VERSION
+    "1.0.0"
+    URL
+    "https://github.com/headupinclouds/csv-parser-cplusplus/archive/v1.0.0.tar.gz"
+    SHA1
+    62eac8c3c82826731cb92556f07eaa56be231ad7
+)
+
+# Pick a download scheme
+hunter_pick_scheme(DEFAULT url_sha1_no_install) # (for cmake modules and header-only libraries)
```
Comment (ruslo) [inline cmake/projects/csv-parser-cplusplus/hunter.cmake]: See https://github.com/headupinclouds/csv-parser-cplusplus/pull/1

File: cmake/projects/cvmatio/hunter.cmake [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+endif()
+
+# Load used modules
+include(hunter_add_version)
+include(hunter_add_version_start)
+include(hunter_download)
+include(hunter_pick_scheme)
+include(hunter_add_package)
+
+hunter_add_package(OpenCV)
```
Comment (ruslo) [inline cmake/projects/cvmatio/hunter.cmake]: This line not needed. This dependency can be managed in `cvmatio` itself. Let's add OpenCV to Hunter first and I will send you pull request to `cvmatio`.

File: cmake/projects/Eigen/hunter.cmake (hunk 1 above)
Comment (headupinclouds) [inline cmake/projects/Eigen/hunter.cmake]: I forgot I added that one.  I'll add the copyright.

File: cmake/projects/cvmatio/hunter.cmake (hunk 4 above)
Comment (headupinclouds) [inline cmake/projects/cvmatio/hunter.cmake]: This is probably too niche to bother with (especially given opencv complications).  I didn't see any easy way to exclude that one.

**Comment by ruslo** [comment]
@headupinclouds Hi, David. Current status:

* `Eigen` - DONE
* `OpenCV` - DONE
* `CsvParserCPlusPlu`s - DONE
* `cvmatio` - wait for patched OpenCV (in progress by me)
* `dlib` - is this library header-only? I see some sources here: https://github.com/davisking/dlib/blob/master/dlib/CMakeLists.txt
* `GLog` - looks fine, can you make a separate pull request please?

**Comment by headupinclouds** [comment]
Thanks for the update.

dlib is header only, but also probably has a limited audience.  What I have
seems to be working fine.
If you'd like it to live in hunter I can push it up.

Same goes for cvmatio (but even more so).  This is really just some code I
needed for matlab and opencv interoperability,
and might just be clutter in your central hunter lib.  I'm guessing there
are < 12 people who have used this code.

Glog is probably useful.  I've been wanting to adopt a "real" stack aware
logging library  (I often end up doing some of this in piecemeal fashion),
but haven't done much with this yet.  I'm not sure how this will do on iOS
and Android, but I can push it up and will get some time to
experiment with it at some point.  If you have any experience with glog or
similar libs I'd be interested.  This is also non-cmake and
I ended up forking a CMake patched fork.  Being a non CMake project  it
might require some additional maintenance to keep the
library up to date with the real upstream code base.  Probably something to
consider.
(It is also not so git friendly, being a google code svn project.)

On that note, I'm also interested in GTest.  I saw you have support for
that so I'll probably take a look soon.
I'm most interested in native C++ cross platform unit tests, in particular
on ARM devices, in order to make sure
ARM specific image processing code behaves similarly to my OS X development
platform.  It wasn't clear if
native iOS GTest functionality was supported, or if this was just for the
simulator.

**Comment by ruslo** [comment]
> If you'd like it to live in hunter I can push it up

The only one problem I can see it's a size of 'hunter.tar.gz' archive. Since Hunter do not hold any binaries it's just a bunch of similar text files. Current version is about 155 KB and 40% of this is wiki/examples/tests + versions of packages that nobody will ever use. So I guess we have pretty big reserve :) In short: I'm okay with any libraries.

> I'm guessing there are < 12 people who have used this code

This is about 6 times larger than number of Hunter users, so it's quite popular.

> dlib is header only

So it's just all '*.h' files from [dlib](https://github.com/davisking/dlib/tree/master/dlib) folder?

> I'm not sure how this will do on iOS and Android

It's not a problem too. It's hard to find the library that just works everywhere. So we can start with some platforms and adopt the code later. If you know that package has a problems with build you can use module `hunter_report_broken_package` ([example](https://github.com/ruslo/hunter/blob/43b22fcd6604cdddc5a0d4c17520df67404ffab9/cmake/projects/GSL/hunter.cmake#L17)). Also you can add notes to wiki ([boost#bugs](https://github.com/ruslo/hunter/wiki/pkg.boost#bugs)).

> being a google code svn project

As far as I know google code is closing soon. So I guess main development will move to github

> It wasn't clear if native iOS GTest functionality was supported, or if this was just for the simulator

I've tested it on both real device and simulator. I just make an executable, open Xcode and run it. Have no idea about how to do it in CI fashion on real device (simulator works fine).

**Pull request closed without merging**